tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,286 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+                                     SIZE_T(dllPathUtf16Len + 1), // +1 for `\0`
+                                     DWORD(MEM_COMMIT),
+                                     DWORD(PAGE_READWRITE))
+      if dllPathRemote == LPVOID(bitPattern: 0) {
+        print("VirtualAllocEx failed \(GetLastError())")
+        return
+      }
+      if !WriteProcessMemory(self.process, dllPathRemote, $0,
+                             SIZE_T(dllPathUtf16Len), nil) {
+        print("WriteProcessMemory failed \(GetLastError())")
+        VirtualFreeEx(self.process, dllPathRemote, 0, DWORD(MEM_RELEASE))
+        return
+      }
+      dllPathWritten = true
+    }
+    if !dllPathWritten {
+      print("Failed to write dll path in the remote process")
+      return nil
+    }
+    return dllPathRemote
+  }
+
+  private func unloadDllAndPathRemote(
+      dwProcessId: DWORD, hKernel32: HMODULE!,
+      dllPathRemote: UnsafeMutableRawPointer) -> Bool {
+    // Get the dll module handle in the remote process to use it to
+    // unload it below.
+    // GetExitCodeThread returns a DWORD (32-bit) but the HMODULE
+    // returned from LoadLibraryW is a 64-bit pointer and may be truncated.
+    // So, search for it using the snapshot instead.
+    var hDllModule = HMODULE(bitPattern: 0)
+    let hModuleSnapshot: HANDLE =
+        CreateToolhelp32Snapshot(DWORD(TH32CS_SNAPMODULE), dwProcessId)
+    if hModuleSnapshot == INVALID_HANDLE_VALUE {
+      print("CreateToolhelp32Snapshot failed")
+      return false
+    }
+    var entry: MODULEENTRY32W = MODULEENTRY32W()
+    entry.dwSize = DWORD(MemoryLayout<MODULEENTRY32W>.size)
+    if !Module32FirstW(hModuleSnapshot, &entry) {
+      print("Module32FirstW failed")
+      return false
+    }
+    repeat {
+      let module: String = withUnsafePointer(to: entry.szModule) {
+        $0.withMemoryRebound(
+          to: WCHAR.self,
+          capacity: MemoryLayout.size(ofValue: $0) / MemoryLayout<WCHAR>.size) {
+          String(decodingCString: $0, as: UTF16.self)
+        }
+      }
+      if module == "SwiftInspectClient.dll" {
+        hDllModule = entry.hModule
+      }
+    } while Module32NextW(hModuleSnapshot, &entry)
+    CloseHandle(hModuleSnapshot)
+    if hDllModule == HMODULE(bitPattern: 0) {
+      print("Failed to find the dll")
+      return false
+    }
+
+    // Unload the dll from the remote process
+    let unloadHookAddr = unsafeBitCast(GetProcAddress(hKernel32, "FreeLibrary"),
+                                       to: LPTHREAD_START_ROUTINE.self)
+    let hUnloadThread = CreateRemoteThread(
+        self.process, nil, 0, unloadHookAddr,
+        unsafeBitCast(hDllModule, to:LPVOID.self), 0, nil)
+    if hUnloadThread == HANDLE(bitPattern: 0) {
+      print("CreateRemoteThread for unload failed \(GetLastError())")
+      return false
+    }
+    defer { CloseHandle(hUnloadThread) }
+    let unload_wait = WaitForSingleObject(hUnloadThread, DWORD(WAIT_TIMEOUT_MS))
+    if unload_wait != WAIT_OBJECT_0 {
```

Comment:
I'd use `guard` and remove the variable.

Reply:
I'd like to keep it as it'd indicate the reason for the return